api/app/utils/wps_dataset.py

```diff
@@ -0,0 +1,188 @@
+from osgeo import gdal, osr
+import numpy as np
+
+from app.utils.geospatial import GDALResamplingMethod
+
+
+class WPSDataset:
+    """
+    A wrapper around gdal datasets for common operations
+    """
+
+    def __init__(self, ds_path: str, band=1, chunk_size=256, size=gdal.GDT_Byte):
+        self.ds = None
+        self.ds_path = ds_path
+        self.band = band
+        self.chunk_size = chunk_size
+        self.size = size
```

Comment:
would `datatype` be more appropriate than `size`?

Reply:
Yes, thank you!